src/pydicom/pixels/decoders/base.py

```diff
@@ -1441,8 +1477,26 @@ def _as_buffer_native(runner: DecodeRunner, index: int | None) -> Buffer:
             same type as in the buffer containing the pixel data unless
             `view_only` is ``True`` in which case a :class:`memoryview` of the
             original buffer will be returned instead.
+
+        Notes
+        -----
+        For certain images, those with BitsAllocated=1, multiple frames and
+        number of pixels per frame that is not a multiple of 8, it is not
+        possible to isolate a buffer to a single frame. A ``RuntimeError`` will
+        be raised in these cases.
         """
         length_bytes = runner.frame_length(unit="bytes")
+
+        if (
+            not isinstance(length_bytes, int)
+            and index is not None
+            and runner.number_of_frames > 1
+        ):
+            raise RuntimeError(
+                "Cannot return a buffer for individual frames since frame "
+                "boundaries do not align with byte boundaries."
+            )
```

Comment:
The Big Endian OB as OW is a similar issue, the expectation with the buffer methods is that the user is responsible for handling issues like this. I'd like it to be supported as-is (i.e. with misaligned frame start/end in the buffer), just add another paragraph to the *Returns* section in `as_buffer()` and `iter_buffer()` explaining the issue with *Bits Allocated* 1

Reply:
Addressed in [896c6b1](https://github.com/pydicom/pydicom/pull/2140/commits/896c6b1fc48f431d63e60d944aedd6d53a947293)